src/SBT.sol

```diff
@@ -1,88 +1,90 @@
-    //////////////////////////////////////////////////////////////*/
+import "./ISoulbound.sol";
 
-    event IssueSingle(
-        address indexed _issuer,
-        address indexed _issuee,
-        uint256 _tokenID,
-        uint256 _value
-    );
-
-    event IssueBatch(
-        address indexed _issuer,
-        address indexed _issuee,
-        uint256[] _tokenIDs,
-        uint256[] _values
-    );
-
-    event RevokeSingle(
-        address indexed _revoker,
-        address indexed _revokee,
-        uint256 _tokenID,
-        uint256 _value
-    );
-
-    event RevokeBatch(
-        address indexed _revoker,
-        address indexed _revokee,
-        uint256[] _tokenIDs,
-        uint256[] _values
-    );
+import {ERC1155} from "openzeppelin-contracts/token/ERC1155/ERC1155.sol";
+import {Ownable} from "openzeppelin-contracts/access/Ownable.sol";
+
+/// @dev See {ISoulbound}.
+abstract contract SBT is ISoulbound, ERC1155, Ownable {
+    //
 
     /*//////////////////////////////////////////////////////////////
                         Views
     //////////////////////////////////////////////////////////////*/
 
-    function hasToken(address _owner, uint256 _tokenID)
+    /// @inheritdoc	ISoulbound
+    function hasToken(address _owner, uint256 _tokenId)
         external
         view
-        returns (bool);
-
-    function hasTokenBatch(
-        address[] calldata _owners,
-        uint256[] calldata _tokenIDs
-    ) external view returns (bool[] memory);
+        returns (bool)
+    {
+        return _hasToken(_owner, _tokenId);
+    }
 
-    function allOwnersOf(uint256 _tokenID)
-        external
+    function _hasToken(address _owner, uint256 _tokenId)
+        internal
         view
-        returns (address[] memory);
+        returns (bool)
+    {
+        return balanceOf(_owner, _tokenId) >= 1;
+    }
 
-    function allTokensOf(address _owner)
+    /// @inheritdoc	ISoulbound
+    function hasTokenBatch(address[] calldata _owners, uint256 _tokenId)
         external
         view
-        returns (uint256[] memory);
+        returns (bool[] memory)
+    {
+        bool[] memory hasTokens = new bool[](_owners.length);
+
+        for (uint256 i = 0; i < _owners.length; i++) {
+            hasTokens[i] = _hasToken(_owners[i], _tokenId);
+        }
+
+        return hasTokens;
+    }
 
     /*//////////////////////////////////////////////////////////////
-                        Issuing
+                        Transactions – Issuing
     //////////////////////////////////////////////////////////////*/
 
-    function issue(address _recipient, uint256 _tokenID) external;
+    /// @inheritdoc	ISoulbound
+    function issue(address _recipient, uint256 _tokenId) external onlyOwner {
+        _issue(_recipient, _tokenId);
+    }
 
-    function issueBatch(
-        address[] calldata _recipients,
-        uint256[] calldata _tokenIDs
-    ) external;
+    function _issue(address _recipient, uint256 _tokenId) internal {
```

Comment:
The pattern of _issue doing the actual work for issue and issueBatch makes sense but was non-obvious at first.  Some doc would be helpful:
issue -- wrapper to issue a single token
issueBatch -- wrapper to issue a batch of tokens to a list of recipients
_issue -- logic to issue a single token, as part of individual issue call or batch